### PR TITLE
Compatibility for upcoming React v16

### PR DESCRIPTION
**Pull request by alexkb0009**
Edits to make library compatible with upcoming React v.16.

Summary:
- Changing Tabs component to be created w/ ES6 class.
- Replacing React.createClass with createReactClass a la https://facebook.github.io/react/blog/2017/04/07/react-v15.5.0.html
- Replacing React.PropTypes with import from https://www.npmjs.com/package/prop-types

Requirements:
- New NPM dependencies 'prop-types' & 'create-react-class'; latest versions (15.5.8, 15.5.2 respectively) OK.

Remainder items (unlikely to be broken for v16 but deprecated): 
- Remove mixins (deprecated),
- Convert remaining React class constructors (TabPane, ScrollableInkTabBar, etc.) from createReactClass to ES6.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,6 +56,7 @@
     "pre-commit": "1.x",
     "rc-tools": "5.x",
     "react": "15.x",
+    "prop-types" : "15.x",
```
Comment (afc163) [inline package.json]: These two should be put in dependencies.

**Comment by afc163** [comment]
Make sure travis ci is success.

**Comment by afc163** [comment]
@alexkb0009

**Comment by afc163** [comment]
Fixed in master, wait for:

- https://github.com/JedWatson/react-hammerjs/pull/67
- https://github.com/JedWatson/react-hammerjs/issues/66

File: src/Tabs.js [hunk 2]
```diff
@@ -16,31 +17,19 @@ function getDefaultActiveKey(props) {
...
+
+  constructor(props){
+    super(props);
+    this.render = this.render.bind(this);
+    this.componentWillReceiveProps = this.componentWillReceiveProps.bind(this);
+    this.onTabClick = this.onTabClick.bind(this);
+    this.onNavKeyDown = this.onNavKeyDown.bind(this);
+    this.setActiveKey = this.setActiveKey.bind(this);
+    this.getNextActiveKey = this.getNextActiveKey.bind(this);
+    this.onTabClick = this.onTabClick.bind(this);
```
Comment (paranoidjk) [inline src/Tabs.js]: No need to repeat bind.

File: src/Tabs.js [hunk 3]
```diff
@@ -16,31 +17,19 @@ function getDefaultActiveKey(props) {
...
-      onChange: noop,
-      tabBarPosition: 'top',
-      style: {},
-    };
-  },
+export default class Tabs extends React.Component {
+
+  constructor(props){
+    super(props);
+    this.render = this.render.bind(this);
```
Comment (paranoidjk) [inline src/Tabs.js]: no need to bind lifecycle function

File: src/Tabs.js [hunk 4]
```diff
@@ -16,31 +17,19 @@ function getDefaultActiveKey(props) {
...
+    this.componentWillReceiveProps = this.componentWillReceiveProps.bind(this);
+    this.onTabClick = this.onTabClick.bind(this);
```
Comment (paranoidjk) [inline src/Tabs.js]: prefer to use arrow function to bind this

File: src/Tabs.js (hunk 2 above)
Comment (paranoidjk) [inline src/Tabs.js]: cc @pingan1927

**Pull request merged**